src/LogicalCalculator.tsx

```diff
@@ -0,0 +1,206 @@
+        setRankPointIndex(e.target.value);
+    }
+
+    const minValue = Math.min(startIndex, endIndex);
+    const maxValue = Math.max(startIndex, endIndex);
+
+    if ((startColumnType === "battery_power" && endColumnType === "ram") || (startColumnType === "ram" && endColumnType === "battery_power")) {
+
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].battery_power);
+            rangeArray.push(data[i].px_height);
+            rangeArray.push(data[i].ram);
+        }
+    }
+
+    if ((startColumnType === "battery_power" && endColumnType === "px_height") || (endColumnType === "battery_power" && startColumnType === "px_height")) {
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].battery_power);
+            rangeArray.push(data[i].px_height);
+        }
+    }
+
+    if ((startColumnType === "px_height" && endColumnType === "ram") || (endColumnType === "px_height" && startColumnType === "ram")) {
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].px_height);
+            rangeArray.push(data[i].ram);
+        }
+    }
+
+    if (startColumnType === "battery_power" && endColumnType === "battery_power") {
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].battery_power);
+        }
+    }
+
+    if (startColumnType === "px_height" && endColumnType === "px_height") {
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].px_height);
+        }
+    }
+
+    if (startColumnType === "ram" && endColumnType === "ram") {
+        for (let i = minValue - 1; i < maxValue; i++) {
+            rangeArray.push(data[i].ram);
+        }
+    }
+
+    const calculateSum = (array: number[]) => {
+        let sum = 0;
+        for (let i = 0; i < array.length; i++) {
+            sum += array[i];
+        }
+        return sum;
+    }
+
+    let rankPointedValue: number;
+    
+    if (rankPointType === "battery_power") {
+        rankPointedValue = data[rankPointIndex - 1].battery_power;
+    } else if (rankPointType === "px_height") {
+        rankPointedValue = data[rankPointIndex - 1].px_height;
+    } else if (rankPointType === "ram") {
+        rankPointedValue = data[rankPointIndex - 1].ram;
+    }
+
+    const calculateAvg = (array: number[]) => {
+        let length = array.length;
+        return Number((calculateSum(array) / length).toFixed(2));
+    }
+
+    const ranking = (arr: number[], compFn: (a: number, b: number) => boolean) => arr.map(a => arr.filter(b => compFn(a, b)).length + 1);
+
+    const calculateRank = (array: number[]) => {
+        const filteredArray: number[] = ranking(array, (a, b) => a < b);
+        const index = rangeArray.indexOf(rankPointedValue);
+        return filteredArray[index];
+    }
+
+    const calculate = () => {
+        switch (operation) {
+            case "MIN":
```

Comment:
Similiar as column types, would be better to make a separated type.
OPERATION.MIN
OPERATION.MAX
...